DAO.sol

```diff
@@ -516,7 +521,12 @@ contract DAO is DAOInterface, Token, TokenCreation {
         uint waitPeriod = p.newCurator
             ? splitExecutionPeriod
             : executeProposalPeriod;
-        // If we are over deadline and waiting period, assert proposal is closed
+                
+        // Add grace period to wait time if proposal appears passable
+        if (p.yea > p.nay)
+           waitPeriod += splittingGracePeriod;
```

Comment:
An extra check could be added to the if statement at line 526 to avoid any possible attacks or vulnerabilities of waitPeriod being summed over and over again, although I don't see the possibility of such an attack- 
waitPeriod == executeProposalPeriod